src/main/java/io/seqera/migtool/Dialect.java

```diff
@@ -0,0 +1,60 @@
+package io.seqera.migtool;
+
+import jakarta.annotation.Nullable;
+import java.util.List;
+
+public enum Dialect {
```

Comment:
Not sure what value adds `Dialect` and `Driver` as enum. it would make more sense to implement a driver pattern that can both the type and the method specialisation for the statement specific for postgres. Likely it would require a `Driver` interface, plus an abstract implementation for the common stuff, and concrete class for each support DB. 

An alternative approach could be using a delegate pattern and passing the concrete class as "argument"